### PR TITLE
Adding loader for DCASE 2023 Task-2

### PR DESCRIPTION
**Pull request by tanmayy24**
### Title

 Please use the following title: "Adding loader for MyDATASET". If your pull request is work in progress, change your title to "[WIP] Adding loader for MyDATASET" to avoid reviews while the loader is not ready.

### Description

Please include the following information at the top level docstring for the dataset's module mydataset.py:

- [x] Describe annotations included in the dataset
- [x] Indicate the total duration of the dataset in hours, and (optionally) also list the number of individual files
- [x] Mention the origin of the dataset (e.g. creator, institution)
- [x] Describe the type of audio included in the dataset
- [x] Indicate any relevant papers related to the dataset
- [x] Include a description about how the data can be accessed and the license it uses (if applicable)

#### Dataset loaders checklist:

- [x] Create a script in `scripts/`, e.g. `make_my_dataset_index.py`, which generates an index file.
- [x] Run the script on the canonical version of the dataset and save the index in `soundata/indexes/` e.g. `my_dataset_index.json`.
- [x] Create a module in soundata, e.g. `soundata/my_dataset.py`.
- [x] Create tests for your loader in `tests/`, e.g. `test_my_dataset.py`.
- [x] Add your module to `docs/source/soundata.rst` and `docs/source/quick_reference.rst`.
- [x] Run `black`, `flake8` and `mypy` (see [Running your tests locally](https://soundata.github.io/en/latest/source/contributing.html#running-your-tests-locally)).
- [x] Run `tests/test_full_dataset.py` on your dataset.
- [x] Check that codecov coverage does not decrease.

If your dataset **is not fully downloadable** there are two extra steps you should follow:
- [x] Contacting the soundata organizers by opening an issue or PR so we can discuss how to proceed with the closed dataset.
- [x] Show that the version used to create the checksum is the "canonical" one, either by getting the version from the dataset creator, or by verifying equivalence with several other copies of the dataset. 
- [x] Make sure someone has run `pytest -s tests/test_full_dataset.py --local --dataset my_dataset` once on your dataset locally and confirmed it passes.

#### Please-do-not-edit flag
To reduce friction, we will make commits on top of contributor's pull requests by default unless they use the `please-do-not-edit` flag. If you don't want this to happen don't forget to add the flag when you start your pull request.

### CODE REVIEW

**Comment by guillemcortes** [comment]
Hi! Is it ready to review? @tanmayy24

**Comment by tanmayy24** [comment]
Yes! It's ready for review now!

File: docs/source/soundata.rst [hunk 1]
```diff
@@ -19,6 +19,13 @@ Dataset Loaders
 .. automodule:: soundata.datasets.marco
    :members:
    :inherited-members:
+
+DCASE23-Task2
+^^^^^^^^
```
Comment (genisplaja) [inline docs/source/soundata.rst]: Finish line!

File: soundata/datasets/dcase23_task2.py [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+            # Paths for metadata files
+            metadata_add_train_path = os.path.join(
+                self.data_home, "7830345", machine, "attributes_00.csv"
+            )
+
+            # Check for file existence
+            if not os.path.exists(metadata_add_train_path):
+                raise FileNotFoundError(
+                    f"Additional training metadata for {machine} not found. Did you run .download()?"
+                )
```
Comment (genisplaja) [inline soundata/datasets/dcase23_task2.py]: Could we test that?

**Review by genisplaja — APPROVED** [review]
Again, thanks @tanmayy24, looks good to me! Minor changes requested, basically a minor documentation thing and a missing test, let's see if it can be added. The rest looks good to me!

**Pull request merged**